chemprop/cli/train.py

```diff
@@ -652,6 +615,58 @@ def main(args):
         args.max_lr,
         args.final_lr,
     )
+
+
+def main(args):
+    format_kwargs = dict(
+        no_header_row=args.no_header_row,
+        smiles_cols=args.smiles_columns,
+        rxn_cols=args.reaction_columns,
+        target_cols=args.target_columns,
+        ignore_cols=args.ignore_columns,
+        weight_col=args.weight_column,
+        bounded=args.loss_function is not None and "bounded" in args.loss_function,
+    )
+    featurization_kwargs = dict(
+        features_generators=args.features_generators, keep_h=args.keep_h, add_h=args.add_h
+    )
+
+    all_data = build_data_from_files(
+        args.data_path,
+        p_features=args.features_path,
+        p_atom_feats=args.atom_features_path,
+        p_bond_feats=args.bond_features_path,
+        p_atom_descs=args.atom_descriptors_path,
+        **format_kwargs,
+        **featurization_kwargs,
+    )
+
+    multicomponent = len(all_data) > 1
+    if not multicomponent:
+        all_data = all_data[0]
```

Comment:
Why not put this in `build_splits()` and rename it `build_data()`? My reasoning to do that is that `all_data` is only used as an input to `build_splits()` and `build_data_from_files()` is already called in `build_splits()` if a separate val or test path is given. This would group all times `build_data_from_files()` is possibly called.

One reason to not put this with the data splitting is it is data loading. But then also maybe building the separate val and test sets would thematically fit more with data loading that data splitting. I feel like I am also splitting hairs here, so I would take the PR as is here.

Reply:
See my other comments below.